utils/document.js

```diff
@@ -0,0 +1,97 @@
+// --- HELPER FUNCTIONS ---
+
+// Return datapoints matching a schema id.
+// @param {Object} content - the annotation content tree (see https://api.elis.rossum.ai/docs/#annotation-data)
+// @param {string} schemaId - the field's ID as defined in the extraction schema(see https://api.elis.rossum.ai/docs/#document-schema)
+// @returns {Array} - the list of datapoints matching the schema ID
+
+const findBySchemaId = (content, schemaId) =>
+    content.reduce(
+        (results, dp) =>
+            dp.schema_id === schemaId ? [...results, dp] :
+                dp.children ? [...results, ...findBySchemaId(dp.children, schemaId)] :
+                    results,
+        []
+    );
+
+// Return datapoints matching an array of schema ids.
+// @param {Object} content - the annotation content tree (see https://api.elis.rossum.ai/docs/#annotation-data)
+// @param {string} schemaIds - an array of field's IDs as defined in the extraction schema(see https://api.elis.rossum.ai/docs/#document-schema)
+// @returns {Object} - key is the schema id, value is the list of datapoints matching the schema ID
+
+const findBySchemaIdBulk = (content, schemaIds) =>
+    schemaIds.reduce(
+        (results, schemaId) => {
+            results[schemaId] = findBySchemaId( content, schemaId );
+            return results;
+        },
+        {
+            valueOf: function (schemaId) { return schemaId in this ? this[schemaId][0].content.value : undefined; },
```

Comment:
These methods are actually essential to use the "inBulk" methods. I added an example to the Webhooks.js

Reply:
I agree that the this keyword can be very confusing, but I could not come up with any other way how to reference "this" object from within the object.

Reply:
Got it. 
An alternative approach would be to treat the bulk as plain methodless object, with these methods as separated helper functions. 
If the original solution sits better with you, consider using arrow function instead of `function` keyword as arrow functions reference `this` based on the context of declarations (where `function` references this based on context of invocation)

Reply:
I see what you mean. The problem with the arrow declaration is that the `this` command would be referencing the initial empty object and the function will always return `undefined` . Of course I could make the object mutable (that was the original implementation), put the result into the results and forget the spread operator, but I assume you won't like this approach either. To summarise my point: yes I need to use `this` in the context of invocation.

Of course, I could also write separate functions and pass the data as an argument. But there is a value in object oriented approach with objects providing methods. Javascript also has `listObject.push( data )` instead of `listMethods.push( listObject, data )`

Reply:
Let's use the mutable approach for now, since this review is already taking long enough. However, for future I'd prefer if we stick to immutable approach, where possible.